src/eigen/lobpcg_hyper_impl.jl

```diff
@@ -324,8 +326,10 @@ function final_retval(X, AX, BX, resid_history, niter, n_matvec)
         BX = BX[:, p]
         resid_history = resid_history[p, :]
     end
+    # compute norms of column vectors with array operations for GPU performance
+    norms = vec(sqrt.(sum(abs2, residuals; dims=1)))
     (; λ=λ_device, X, AX, BX,
-     residual_norms=norm.(eachcol(residuals)),
+     residual_norms=oftype(ones(eltype(norms), 1), norms),
```

Comment:
Add a `to_cpu` method?

Reply:
Valid point, it would make a lot of sense here